reflekt-plugin/src/main/kotlin/org/jetbrains/reflekt/plugin/generation/ir/StorageClassGenerator.kt

```diff
@@ -0,0 +1,178 @@
+    fun contributeInitializer(storageClassSymbol: IrClassSymbol, storedClassesSymbols: Collection<IrClassSymbol>) {
+        val storageClass = storageClassSymbol.owner
+        val storedClasses = storedClassesSymbols.map { it.owner }
+
+        // Adding an anonymous initializer filling storage field with data from storedClassesSymbols.
+        storageClass.contributeAnonymousInitializer {
+            // At the first, ReflektClassImpl is instantiated for each stored class without superclasses and sealed subclasses data and is stored to HashMap.
+            val mVariable = irVariableVal(
+                parent = storageClass,
+                name = mVariableName,
+                type = generationSymbols.hashMapClass.createType(
+                    false,
+                    listOf(irBuiltIns.kClassClass.starProjectedType, generationSymbols.reflektClassClass.starProjectedType)
+                ),
+                isConst = false,
+                isLateinit = false,
+            ).also { variable ->
+                variable.initializer = irHashMapOf(
+                    keyType = irBuiltIns.kClassClass.starProjectedType,
+                    valueType = generationSymbols.reflektClassClass.starProjectedType,
+                    pairs = storedClasses.map { storedClass ->
+                        irTo(left = irClassReference(storedClass.symbol), right = irReflektClassImplConstructor(storedClass.symbol))
+                    },
+                )
+            }
+            +mVariable
+            // Then, add calls are generated to set up superclasses and sealed subclasses for each stored class.
+            for (storedClass in storedClasses) {
+                storedClass.getImmediateSuperclasses().forEach { superclass ->
+                    +irMutableSetAdd(
+                        mutableSet = irCall(
+                            generationSymbols.reflektClassImplGetSuperclasses,
+                            dispatchReceiver = irTypeCast(
+                                generationSymbols.reflektClassImplClass.createType(false, listOf(storedClass.defaultType)),
+                                irCheckNotNull(irMapGet(map = irGet(mVariable), key = irClassReference(storedClass.symbol))),
+                            ),
+                        ),
+                        element = irTypeCast(
+                            generationSymbols.reflektClassClass.createType(
+                                false,
+                                listOf(makeTypeProjection(storedClass.defaultType, Variance.IN_VARIANCE)),
+                            ),
+                            irCheckNotNull(
+                                value = irMapGet(
+                                    map = irGet(mVariable),
+                                    key = irClassReference(superclass),
+                                ),
+                            ),
+                        ),
+                    )
+                }
+            }
```

Comment:
It looks almost the same with 140-168 code rows, I think we can move them into a function and just call it here